pkg/provider/azure_vmssflex.go

```diff
@@ -989,7 +989,9 @@ func (fs *FlexScaleSet) ensureBackendPoolDeletedFromNode(vmssFlexVMNameMap map[s
 			nics[nicName] = nic
 		}
 	}
+	nicUpdaterLock := "nicUpdaterLock"
```

Comment:
+1 atomic.Store() may be better than lock?

Reply:
Will update in the other PR and rebase.